src/services/CertificateGenerationService.ts

```diff
@@ -1,898 +1,900 @@
-   */
-  public async getVehicleMakeAndModel(testResult: any) {
-    const techRecord = await this.getTechRecord(testResult);
-    // Return bodyMake and bodyModel values for PSVs
-    if (techRecord.techRecord[0].vehicleType === VEHICLE_TYPES.PSV) {
-      return {
-        Make: techRecord.techRecord[0].chassisMake,
-        Model: techRecord.techRecord[0].chassisModel,
-      };
-    } else {
-      // Return make and model values for HGV and TRL vehicle types
-      return {
-        Make: techRecord.techRecord[0].make,
-        Model: techRecord.techRecord[0].model,
-      };
+        return {
+            currentRecords,
+            provisionalRecords,
+            currentCount: currentRecords.length,
+            provisionalCount: provisionalRecords.length
+        };
     }
-  }
-
-  /**
-   * Method for getting techRecord to which the test-results reffer to
-   * @param testResult - the testResult for which the tech record search is done for
-   */
-  public async getTechRecord(testResult: any) {
-    let techRecords: any | any[] = testResult.systemNumber
-      ? await this.queryTechRecords(testResult.systemNumber, "systemNumber")
-      : undefined;
-    if (!isSingleRecord(techRecords) && testResult.vin) {
-      console.log(
-        "No unique Tech Record found for systemNumber ",
-        testResult.systemNumber,
-        ". Trying vin"
-      );
-      techRecords = await this.queryTechRecords(testResult.vin);
+
+    /**
+     * Returns true if an adrDetails object contains a tankDetails object
+     * @param testResult - testResult from which the VIN is used to search a tech-record
+     */
+    public async containsTankDetails(adrDetails: any) {
+        return adrDetails && !!await this.unflattenFlatObject(adrDetails, "tank") && !!await this.unflattenFlatObject(adrDetails, "tankDetails");
     }
-    if (!isSingleRecord(techRecords) && testResult.partialVin) {
-      console.log(
-        "No unique Tech Record found for vin ",
-        testResult.vin,
-        ". Trying Partial Vin"
-      );
-      techRecords = await this.queryTechRecords(testResult.partialVin);
+
+    /**
+     * Retrieves the vehicle weight details for Roadworthisness certificates
+     * @param testResult
+     */
+    public async getWeightDetails(testResult: any) {
+        const searchRes = await this.callSearchTechRecords(testResult.vin);
+        const techRecord = await this.processGetCurrentProvisionalRecords(searchRes) as TechRecordType<"hgv" | "psv" | "trl">;
+        if (techRecord) {
+            const weightDetails: IWeightDetails = {
+                dgvw: techRecord.techRecord_grossDesignWeight ?? 0,
+                weight2: 0,
+            };
+            if (testResult.vehicleType === VEHICLE_TYPES.HGV) {
+                weightDetails.weight2 = (techRecord as TechRecordType<"hgv">).techRecord_trainDesignWeight ?? 0;
+            } else {
+                if (
+                    techRecord.techRecord_noOfAxles ?? -1 > 0
```

Comment:
What is this line trying to do?
If `techRecord_noOfAxles` is nullish, the second part of that expression always evaluates to `false`. I'm assuming that's not the intended behaviour.

I'm assuming this is what's intended?
```suggestion
                    (techRecord.techRecord_noOfAxles ?? -1) > 0
```
might be clearer to use:
```suggestion
                    techRecord.techRecord_noOfAxles && techRecord_noOfAxles > 0
```